tests/integration/hsr_test.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import pytest
+
+import libnmstate
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import Hsr
+
+from .testlib import assertlib
+
+
+ETH1 = "eth1"
+ETH2 = "eth2"
+HSR0 = "hsr0"
+
+
+@pytest.mark.tier1
+def test_add_hsr_and_remove(eth1_up, eth2_up):
```

Comment:
Does NM always support the hsr/prp interface? maybe the tests need to add certain condition about which NM version the test should be based on.

Reply:
I also find similar failures when I apply the example manually:
```
[root@fcvm rust]# ./target/debug/nmstatectl apply ../examples/hsr0_up.yml 
[2023-12-14T21:58:08Z INFO  nmstate::query_apply::net_state] Created checkpoint /org/freedesktop/NetworkManager/Checkpoint/1
[2023-12-14T21:58:08Z INFO  nmstate::nm::query_apply::profile] Creating connection UUID Some("419ff1b8-138b-4cd3-a9d1-888e9869cdbe"), ID Some("hsr0"), type Some("hsr") name Some("hsr0")
[2023-12-14T21:58:08Z INFO  nmstate::query_apply::net_state] Rollbacked to checkpoint /org/freedesktop/NetworkManager/Checkpoint/1
NmstateError: DependencyError: Please upgrade NetworkManager for specified interface type: Connection(InvalidSetting):hsr: unknown setting name

```

Reply:
Nice catch. Fixed!